python/cudf/cudf/_libxx/parquet.pyx

```diff
@@ -35,6 +40,40 @@ from cudf._libxx.io.utils cimport (
 cimport cudf._libxx.cpp.types as cudf_types
 cimport cudf._libxx.cpp.io.types as cudf_io_types
 
+cdef class BufferArrayFromVector:
+    cdef unsigned length
+    cdef unique_ptr[vector[uint8_t]] in_vec
+
+    # these two things declare part of the buffer interface
+    cdef Py_ssize_t shape[1]
+    cdef Py_ssize_t strides[1]
+
+    cdef set_ptr(self, unique_ptr[vector[uint8_t]] in_vec):
```

Comment:
Per the [cython docs](https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#existing-pointers-instantiation), we can initialize a new `BufferArrayFromVector` instance via a static method like this:
```suggestion
    @staticmethod
    cdef BufferArrayFromVector from_ptr(unique_ptr[vector[uint8_t]] in_vec):
```